diff --git a/data/sol.metainfo.xml.in b/data/sol.metainfo.xml.in
index 25319eb..6502655 100644
--- a/data/sol.metainfo.xml.in
+++ b/data/sol.metainfo.xml.in
@@ -27,9 +27,7 @@
   of the GNU Free Documentation License along with this program.
   If not, see <https://www.gnu.org/licenses/>.
 -->
-<component
-  xmlns="https://specifications.freedesktop.org/metainfo/1.0"
-  type="desktop">
+<component type="desktop-application">
   <id>sol.desktop</id>
   <metadata_license>GPL-3.0+ or GFDL-1.3-only</metadata_license>
   <project_license>GPL-3.0+</project_license>
@@ -52,8 +50,15 @@
     <kudo>ModernToolkit</kudo>
     <kudo>UserDocs</kudo>
   </kudos>
+  <releases>
+    <release version="3.22.33" date="2024-05-18" />
+  </releases>
+  <content_rating type="oars-1.1" />
   <url type="homepage">https://wiki.gnome.org/Apps/Aisleriot</url>
+  <url type="vcs-browser">https://gitlab.gnome.org/GNOME/aisleriot</url>
+  <url type="translate">https://l10n.gnome.org/module/aisleriot/</url>
   <project_group>GNOME</project_group>
+  <developer_name>The GNOME Project</developer_name>
   <translation type="gettext">aisleriot</translation>
-  <launchable type="desktop-id">sol.desktop</launchable>
+  <launchable type="desktop-id">org.gnome.Aisleriot.desktop</launchable>
 </component>
--
libgit2 1.7.2

