--- a/data/sol.appdata.xml.in	2019-09-09 21:51:48.000000000 +0200
+++ b/data/sol.appdata.xml.in	2019-09-27 12:51:25.571094693 +0200
@@ -51,6 +51,10 @@
     <kudo>ModernToolkit</kudo>
     <kudo>UserDocs</kudo>
   </kudos>
+  <releases>
+    <release version="3.22.9" date="2019-09-09" />
+  </releases>
+  <content_rating type="oars-1.0" />
   <url type="homepage">https://wiki.gnome.org/Apps/Aisleriot</url>
   <project_group>GNOME</project_group>
   <translation type="gettext">aisleriot</translation>
